internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -124,7 +141,9 @@ func (r *BMACReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 			r.Log.WithError(err).Errorf("Error updating after BMH reconcile")
 			return reconcileError{err}.Result()
 		}
+	}
 
+	if result.Stop() {
```

Comment:
Each place this is evaluated is also a good and appropriate place to check if the context has been cancelled, to participate in graceful shutdown of the process. WDYT about working that in? It's not a priority, so it could be deferred, but I thought I'd bring it up since this PR is so closely related.

You could do one of:

- make each line like this read `if result.Stop() || ctx.Err() != nil {`
- pass the context to the Stop() function and let it do the evaluation in one place

Reply:
I vote for:
> pass the context to the Stop() function and let it do the evaluation in one place

Reply:
I like it, thanks for the suggestion! :smiley: